src/NUnitFramework/framework/Constraints/DictionaryContainsKeyValuePairConstraint.cs

```diff
@@ -40,12 +43,31 @@ public override string Description
 
         private bool Matches(object actual)
         {
-            var dictionary = ConstraintUtils.RequireActual<IDictionary>(actual, nameof(actual));
-            foreach (var entry in dictionary)
-                if (ItemsEqual(entry, _expected))
-                    return true;
+            if (actual == null)
+                throw new ArgumentException("Expected: IDictionary But was: null", nameof(actual));
 
-            return false;
+            if (TypeHelper.TryCast<IDictionary>(actual, out var dictionary))
+            {
+                foreach (var entry in dictionary)
+                    if (ItemsEqual(entry, _expected))
+                        return true;
+                return false;
+            }
+
+            // If 'actual' implements IDictionary<TKey, TValue>, construct an 'expected' KeyValuePair<TKey, TValue>
+            // and look it up by iterating using IEnumerable
+            if (actual.GetType().GetInterfaces().Any(i => i.FullName.StartsWith("System.Collections.Generic.IDictionary`2[")))
```

Comment:
I'd suggest smth like that instead:
```csharp
i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IDictionary<,>)
```

Reply:
That's a good idea. Just made the change. Thanks @Dreamescaper.